image/descriptor.go

```diff
@@ -49,6 +49,30 @@ func (d *descriptor) hash() string {
 	return pts[1]
 }
 
+func findDescriptors(w walker) (map[string]*descriptor, error) {
+	refs := make(map[string]*descriptor)
+
+	if err := w.walk(func(path string, info os.FileInfo, r io.Reader) error {
+		if info.IsDir() || !strings.HasPrefix(path, "refs") {
+			return nil
+		}
+
+		var d descriptor
+		if err := json.NewDecoder(r).Decode(&d); err != nil {
+			return err
+		}
+		refs[info.Name()] = &d
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	if len(refs) == 0 {
+		return nil, errors.New("no descriptors found")
```

Comment:
This is a _validation_ question, I opened https://github.com/opencontainers/image-spec/issues/287 to discuss more about this one, I'll move this check in `validate`.